[IMP] [16.0] stock_inventory_discrepancy: UI improvements

Shide opened the pull request:
Changes:
- optional="hide" on list view fields
- Discrepancy view group with stock.group_stock_manager
- Company related icon on Settings

![image](https://github.com/user-attachments/assets/7bfd9370-f8e5-44f8-a30d-c5948936c087)
![image](https://github.com/user-attachments/assets/87e477fe-16da-486d-a60b-2ac30bb4a3ea)


MT-9768 @moduon @rafaelbn @yajo @EmilioPascual @LoisRForgeFlow please review if you want 😄

rafaelbn commented:
Hello @LoisRForgeFlow @AaronHForgeFlow @JoanSForgeFlow @JasminSForgeFlow what do you think about this UX improvements? 😄

LoisRForgeFlow commented on stock_inventory_discrepancy/views/stock_location_view.xml:
```diff
@@ -7,13 +7,19 @@
+                <group
+                    name="grp_discrepancy_threshold"
+                    string="Inventory Discrepancy"
+                    groups="base.group_no_one"
```
I agree in general with the improvements. Just a bit hesitant on making them only displayed in debug mode, any specific reason for this?

Shide replied:
There is a general threshold configured in settings, so... 99% of the time, this information is redundant (most of the time) and doesn't change too often.
And other "configurable" fields in stock location, which follows the same kind of usage are displayed on debug mode too.

LoisRForgeFlow replied:
I'm not sure if what you state is correct:

There is no discrepancy threshold in the settings
![image](https://github.com/user-attachments/assets/f19f4cf7-a7f5-4d55-aa16-71fd4c9fba21)

And there is not a single field with group_no_one in standard location form view:
![image](https://github.com/user-attachments/assets/b639e141-aaf0-478e-ba58-d345e97c72f2)


I think I would keep standard visibility for these fields, you can apply the group "group_stock_manager" if you want to restrict it a little bit.

Regarding the WH view, I would move the new group to the wh config tab

![image](https://github.com/user-attachments/assets/b14334d3-187a-4795-af73-5d729b5b1d46)


Is this fine for you?

Shide replied:
Totally 😄 
But checking it again, I've respected the original position (added manager group) because this pages are not shown if you don't activate certain settings.

Shide committed:
[FIX] stock_inventory_discrepancy: UI improvements

- optional="hide" on list view fields
- Discrepancy view group with stock.group_stock_manager
- Company related icon on Settings

LoisRForgeFlow reviewed (approved):
Thanks! LGTM now :+1:

AaronHForgeFlow reviewed (approved):
LGTM :+1:

rafaelbn reviewed (approved):
/ocabot merge minor

LoisRForgeFlow commented:
@Shide could you take care of fwporting these change to next versions?

LoisRForgeFlow commented:
@Shide could you fwport this?

Shide commented:
@LoisRForgeFlow doing in https://github.com/OCA/stock-logistics-warehouse/pull/2346, but it will take me more time
We don't have customers in v17, so I've ported all to v18 (all of this PRs must be ported together)
